boot.mount

```diff
@@ -1,6 +1,5 @@
 [Unit]
 ConditionKernelCommandLine=|ostree
-ConditionPathIsMountPoint=|/sysroot/boot
```

Comment:
We don't really need it as long as the ostree binary keeps doing the bind mount in the initrd. I personally think it is more flexible to have systemd do the bind mount from the rootfs, but I don't see we changing this anytime soon. So I'll add an extra commit removing boot.mount as well.